### DISCUSSION TITLE
Why are curly braces required between InitApi and ShutdownApi?

### DISCUSSION THREAD

**Discussion by buddhike** [aws/aws-sdk-cpp#2897] [Q&A] [answered]
Docs for C++ SDK states the following:

> All AWS SDK for C++ calls performed between Aws::InitAPI and Aws::ShutdownAPI should either to be contained within a pair of curly braces or should be invoked by functions called between the two methods.

What’s the significance of using curly braces here?

**Comment by jmklix** [comment, marked as answer]
This serves the purpose of adding a scope guard to all of the sdk calls. It makes sure that the sdk is correctly and completely initialized before any other sdk calls are make. And it makes sure everything is destroyed and ready to be cleaned up at the end of the curly braces.

**Reply by buddhike** [reply]
Thank you! 🙏🏾
